staging/src/k8s.io/client-go/rest/request_watchlist_test.go

```diff
@@ -0,0 +1,340 @@
+				TypeMeta: metav1.TypeMeta{Kind: "PodList", APIVersion: "v"},
+				Items:    []v1.Pod{},
+			},
+		},
+		{
+			name:   "rv is applied",
+			result: &v1.PodList{},
+			target: WatchListResult{initialEventsEndBookmarkRV: "100"},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{Kind: "PodList"},
+				ListMeta: metav1.ListMeta{ResourceVersion: "100"},
+				Items:    []v1.Pod{},
+			},
+		},
+		{
+			name:   "items are applied",
+			result: &v1.PodList{},
+			target: WatchListResult{items: []runtime.Object{makePod(1), makePod(2)}},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{Kind: "PodList"},
+				Items:    []v1.Pod{*makePod(1), *makePod(2)},
+			},
+		},
+		{
+			name:        "type mismatch",
+			result:      &v1.PodList{},
+			target:      WatchListResult{items: []runtime.Object{makeNamespace("1")}},
+			expectedErr: fmt.Errorf("received object type = v1.Namespace at index = 0, doesn't match the list item type = v1.Pod"),
+		},
+	}
+	for _, scenario := range scenarios {
+		t.Run(scenario.name, func(t *testing.T) {
+			err := scenario.target.Into(scenario.result)
+			if scenario.expectedErr != nil && err == nil {
+				t.Fatalf("expected an error = %v, got nil", scenario.expectedErr)
+			}
+			if scenario.expectedErr == nil && err != nil {
+				t.Fatalf("didn't expect an error, got =  %v", err)
+			}
+			if err != nil {
+				if scenario.expectedErr.Error() != err.Error() {
+					t.Fatalf("unexpected err = %v, expected = %v", err, scenario.expectedErr)
+				}
+				return
+			}
+			if !apiequality.Semantic.DeepEqual(scenario.expectedResult, scenario.result) {
+				t.Errorf("diff: %v", cmp.Diff(scenario.expectedResult, scenario.result))
+			}
+		})
+	}
+}
+
+func TestWatchListSuccess(t *testing.T) {
+	scenarios := []struct {
+		name           string
+		watchEvents    []watch.Event
+		expectedResult *v1.PodList
+	}{
+		{
+			name: "happy path",
+			watchEvents: []watch.Event{
+				{Type: watch.Added, Object: makePod(1)},
+				{Type: watch.Added, Object: makePod(2)},
+				{Type: watch.Bookmark, Object: makeBookmarkEvent(5)},
+			},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{
+					// Note that the APIVersion for the core API group is "v1" (not "core/v1").
+					// We fake "core/v1" here to test if the Group part is properly
+					// recognized and set on the resulting object.
+					APIVersion: "core/v1",
+					Kind:       "PodList",
+				},
+				ListMeta: metav1.ListMeta{ResourceVersion: "5"},
+				Items:    []v1.Pod{*makePod(1), *makePod(2)},
+			},
+		},
+		{
+			name: "only the bookmark",
+			watchEvents: []watch.Event{
+				{Type: watch.Bookmark, Object: makeBookmarkEvent(5)},
+			},
+			expectedResult: &v1.PodList{
+				TypeMeta: metav1.TypeMeta{
+					// Note that the APIVersion for the core API group is "v1" (not "core/v1").
+					// We fake "core/v1" here to test if the Group part is properly
+					// recognized and set on the resulting object.
+					APIVersion: "core/v1",
```

Comment:
Actually about that - can you make group & version to be params of the test? I would like to ensure that we also handle the case of empty group correctly.

So let's maybe use core/v1 in one of the test cases and "v1" in the second one.

Reply:
sure, good idea.

Reply:
done.